tests/prague/eip7480_data_section/test_data_opcodes.py

```diff
@@ -0,0 +1,254 @@
+from ethereum_test_tools.vm.opcode import Opcode
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7480.md"
+REFERENCE_SPEC_VERSION = "3ee1334ef110420685f1c8ed63e80f9e1766c251"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+contract_call_within_deep_nested_callf = Container(
+    name="contract_call_within_deep_nested_callf",
+    sections=[
+        Section.Code(
+            code=(Op.CALLF[1] + Op.PUSH1(1) + Op.PUSH0 + Op.SSTORE + Op.STOP),
+            code_inputs=0,
+            code_outputs=NON_RETURNING_SECTION,
+            max_stack_height=2,
+        )
+    ]
+    + [
+        # All sections call next section and on return, store a 1
+        # to their call stack height key
+        Section.Code(
+            code=(Op.CALLF[i] + Op.PUSH1(1) + Op.PUSH2(i - 1) + Op.SSTORE + Op.RETF),
+            code_inputs=0,
+            code_outputs=0,
+            max_stack_height=2,
+        )
+        for i in range(2, MAX_CODE_SECTIONS)
+    ]
+    + [
+        # Last section makes external contract call
+        Section.Code(
+            code=(
+                Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH2(0x200)
+                + Op.EXTCALL
+                + Op.ISZERO
+                + Op.PUSH2(MAX_CODE_SECTIONS - 1)
+                + Op.SSTORE
+                + Op.RETF
+            ),
+            code_inputs=0,
+            code_outputs=0,
+            max_stack_height=4,
+        )
+    ],
+)
+
+recursive_contract_call_within_deep_nested_callf = Container(
+    name="recursive_contract_call_within_deep_nested_callf",
+    sections=[
+        # All sections call next section and on return, store a 1
+        # to their call stack height key
+        Section.Code(
+            code=(Op.CALLF[i + 1] + Op.PUSH1(1) + Op.PUSH2(i) + Op.SSTORE + Op.STOP),
+            code_inputs=0,
+            code_outputs=NON_RETURNING_SECTION,
+            max_stack_height=2,
+        )
+        for i in range(MAX_CODE_SECTIONS - 1)
+    ]
+    + [
+        # Last section makes external contract call
+        Section.Code(
+            code=(
+                Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH0
+                + Op.PUSH2(0x200)
+                + Op.GAS
+                + Op.CALL
+                + Op.PUSH2(MAX_CODE_SECTIONS - 1)
+                + Op.SSTORE
+                + Op.RETF
+            ),
+            code_inputs=0,
+            code_outputs=0,
+            max_stack_height=7,
+        )
+    ],
+)
+
+
+def create_data_test(name: str, offset: int, datasize: int):
+    """
+    Generates data load operators test cases based on load offset and data section size.
+    """
+    data = "".join("%04x" % i for i in range(1, datasize // 2 + 1))
```

Comment:
```suggestion
    data = b"".join(i.to_bytes(length=2, byteorder="big") for i in range(1, datasize // 2 + 1))
    assert len(data) == datasize
```
I think working with bytes instead of strings makes a bit easier to understand.

Next two comments are related to this one.